block-layer/codecs/DAG-JSON.md

```diff
@@ -24,14 +24,18 @@ Contrary to popular belief, JSON as a format supports Big Integers. It's only
 JavaScript itself that has trouble with them. This means JS implementations
 of `dag-json` can't use the native JSON parser and serializer.
 
+### Other Types
+
+The IPLD Data Model specifies two types that cannot directly be represented in JSON. They are both represented as [Multibase](https://github.com/multiformats/multibase) Base64 encoded strings. The Base64 encoding is the one described in [RFC 4648, section 4](https://tools.ietf.org/html/rfc4648#section-4) without padding, hence the Multibase prefix is `m`.
+
 #### Binary Type
 
 ```javascript
-{"/": { "base64": String }}
+{"/": { "base64": String /* Multibase Base64 encoded binary */ }}
 ```
 
 #### Link Type
 
 ```javascript
-{"/": String /* base encoded CID */}
+{"/": String /* Multibase Base64 encoded CID */}
```

Comment:
Why specify specifically Base64 for this? If only allowing base64 encoding then why include the multibase prefix in the string encoding?

Reply:
Because almost all the code we have to generate a string from a CID returns a multibase encoded string. It would actually be more error prone for this to not be multibase encoded.

Reply:
yeah I agree, I want it to be multibase encoded, I'm just saying for the spec to say in this comment that it is Base64 specifically might be a bit misleading?

Reply:
@ausinabell The idea is that it must be Base64 encoded CID. So that implementations don't need to support all base encodings. You can also generate Base64 encoded strings in the browser.

Reply:
I get that from a compactness standpoint but it would be nice if the encoding matched the default CID encoding (base32).

Also, what about CIDv0?

Reply:
> Also, what about CIDv0?

Ah, I see. Below.

Reply:
Okay gotcha, didn't know there was a limitation for generating any of the multibases in js. Also yeah the default Cid encoding of base32 lower would be nice